src/lib/samples.rs

```diff
@@ -181,6 +177,29 @@ mod tests {
         assert!(samples_metadata.samples[1].barcode == "CATGCTA");
     }
 
+    #[test]
+    fn test_tsv_file_delim_error() {
+        let lines = vec![
```

Comment:
I like the `[...].iter().map(|&s| s.into()).collect()` idiom better when the list of strings grows long, but this is fine if it's only ever going to be a few